src/charm.py

```diff
@@ -614,7 +709,30 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
             f"Migration of sharding components not permitted, revert config role to {self.role}"
         )
 
-    def _on_start(self, event) -> None:
+    def __start_checks(self) -> bool:
```

Comment:
nit

```suggestion
    def __can_charm_start(self) -> bool:
```

functions that return are bool are nice to have a question like name i.e.  starting with `is`, `can`, `are`, etc